roles/openshift_health_checker/test/etcd_traffic_test.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from openshift_checks.etcd_traffic import EtcdTraffic, OpenShiftCheckException
+
+
+@pytest.mark.parametrize('group_names,is_containerized,version,is_active', [
+    (['masters'], False, "3.5", True),
+    (['masters'], True, "3.6", False),
+    (['nodes'], False, "3.6", False),
+    (['nodes'], True, "3.4", False),
+    (['etcd'], False, "3.4", True),
+    (['etcd'], True, "3.5", True),
+    (['etcd'], False, "3.5", True),
+    (['masters', 'nodes'], False, "3.5", True),
+    (['masters', 'etcd'], False, "3.3", False),
+    ([], False, "3.4", False),
+    (['lb'], False, "3.6", False),
+    (['nfs'], False, "3.5", False),
+])
+def test_is_active(group_names, is_containerized, version, is_active):
+    task_vars = dict(
+        group_names=group_names,
+        openshift=dict(
+            common=dict(is_containerized=is_containerized, short_version=version),
+        ),
+    )
+    assert EtcdTraffic.is_active(task_vars=task_vars) == is_active
+
+
+@pytest.mark.parametrize('group_names,matched,failed,master_count,extra_words', [
+    (["masters"], True, False, 1, ["Higher than normal etcd traffic detected"]),
+    (["masters", "etcd"], False, True, 1, ["No log messages matched the expression"]),
+    (["etcd"], False, False, 5, []),
+])
+def test_log_matches_high_traffic_msg(group_names, matched, failed, master_count, extra_words):
```

Comment:
That's a problem we have in general. Of course the tests will catch it if we miss something... probably.

Do you have a suggestion? Generally I would solve this by either introducing an unchanging unique ID in each message (which is more annoying than you might think, see current oadm diagnostics implementation), or turning each message into a record with attrs instead of just a string, or just structuring code and tests in such a way that the test can determine easily which error occurred before it gets translated into something meant for humans.